src/coreclr/jit/codegenxarch.cpp

```diff
@@ -2511,7 +2511,7 @@ void CodeGen::genLclHeap(GenTree* tree)
             amount /= STACK_ALIGN;
         }
 
-        genSetRegToIcon(regCnt, amount, ((int)amount == amount) ? TYP_INT : TYP_LONG);
+        genSetRegToIcon(regCnt, amount, FitsIn<int>(amount) ? TYP_INT : TYP_LONG);
```

Comment:
This is a behavior change on 32-bit platforms when `amount` has the upper bit set. It will make a `TYP_LONG` node after this.
I think the equivalent expression that GCC should not warn about is `(size_t)(int)amount == amount`.

Reply:
> This is a behavior change on 32-bit platforms when `amount` has the upper bit set. 

AFAICT, `FitsIn<T>` (from safemath) should expand to same expression as written before.

Reply:
https://godbolt.org/z/8G7dedrYs (with -m32)

Reply:
Ah right, I can see the difference. :)

Reply:
I don't see any test with `FitsIn<int>`.

Reply:
Updated all callsites with suggested approach.